controllers/challenge.js

```diff
@@ -12,10 +12,17 @@ const sendChallengeResponse = async (req, res) => {
   try {
     if (req.method === 'GET') {
       const allChallenges = await challengeQuery.fetchChallenges()
-      if (allChallenges.length > 0) {
+      const updatedChallengesData = await Promise.all(allChallenges.map(async (challenge) => {
+        const participants = await challengeQuery.fetchParticipantsData(challenge.participants)
+        return {
+          ...challenge,
+          participants
+        }
+      }))
```

Comment:
A lot seems to be happening here, could please separate out `allChallenges.map` part?

Reply:
Wait, why is the function name `updatedChallengesData` inside `GET`?